distutils/util.py

```diff
@@ -151,7 +151,9 @@ def change_root(
         if not os.path.isabs(pathname):
             return os.path.join(new_root, pathname)
         else:
-            return os.path.join(new_root, pathname[1:])
+            # type-ignore: This makes absolutes os.Pathlike unsupported in this branch.
+            # Either this or we don't support bytes-based paths, or we complexify this branch.
+            return os.path.join(new_root, pathname[1:])  # type: ignore[index]
```

Comment:
There's a decision to be taken here.
1. Don't support `PathLike` in this branch (current behaviour)
2. Don't support `bytes | PathLike[bytes]` in this branch
```suggestion
            # type-ignore: This makes absolutes os.Pathlike unsupported in this branch.
            # Either this or we don't support bytes-based paths, or we complexify this branch.
            return os.path.join(new_root, str(pathname)[1:])
```
3. Support both, requires extra checking.